polar-core/src/vm.rs

```diff
@@ -1401,17 +1401,14 @@ impl PolarVirtualMachine {
         }
     }
 
-    pub fn check_error(&self) -> PolarResult<QueryEvent> {
-        if let Some(error) = &self.external_error {
+    pub fn check_error(&mut self) -> PolarResult<QueryEvent> {
+        if let Some(msg) = self.external_error.take() {
```

Comment:
I inspected the other uses of `self.external_error`, and this move seems chill. Could probably use a second pair of 👀

Reply:
Talked offline. @samscott89 "trusts my judgment" — only time will tell how poor his own is